### PR TITLE
Introduce pomodoro utility module

### PR DESCRIPTION
**Pull request by EthanThatOneKid**
### Changelog

- Add pomodoro utility module (`mod.ts`) with tests.
- Define GitHub workflows
  - `.github/workflows/check.yaml`: Checks code formatting, linting, and tests.
  - `.github/workflows/publish.yaml`: Publishes to `npm` branch on push to
    `main` branch.
- Add `README.md` sections
  - Installation and Usage
  - API
  - License

### CODE REVIEW

File: cycle.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+export class Cycle {
```
Comment (EthanThatOneKid) [inline cycle.ts:4]: Provide brief JSDoc comment for the exported `Cycle` class.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,53 @@
...
+```ts
+import { format, Pomo } from "pomo";
+```
+
+### Example usage
+
+```ts
+const pomo = Pomo.fromPattern({
+  pattern: "25w5b25w5b25w5b25w15b",
+  dayLength: 25 * 60 * 1000,
```
Comment (EthanThatOneKid) [inline README.md]: I like `1e3`.
```suggestion
  dayLength: 25 * 60 * 1e3,
```

File: pomo.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import { Cycle } from "./cycle.ts";
+import { parsePattern } from "./pattern.ts";
+import type { PatternOptions } from "./pattern.ts";
+
+export class Pomo {
```
Comment (EthanThatOneKid) [inline pomo.ts:8]: Provide brief JSDoc comment for the exported `Pomo` class.

File: pomo.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+     * break periods.
+     */
+    public readonly cycle: Cycle,
+    /** The length of a day in the same unit as the cycle. */
+    public readonly dayLength: number,
+    /** The number for the cycle to reference. */
+    public readonly ref: number,
+  ) {}
+
+  public get infinite(): boolean {
```
Comment (EthanThatOneKid) [inline pomo.ts]: Provide brief JSDoc comment for the public `infinite` method.

File: pomo.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    return this.cycle.periods.length % 2 === 0 &&
+      this.dayLength % this.cycle.total === 0;
+  }
+
+  public at(n: number): PomoStamp {
```
Comment (EthanThatOneKid) [inline pomo.ts:30]: Provide brief JSDoc comment for the public `at` method.

File: pomo.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    return new PomoStamp(this, n);
+  }
+
+  public static fromPattern(options: PatternOptions): Pomo {
```
Comment (EthanThatOneKid) [inline pomo.ts:53]: Provide JSDoc comment including example usage for the public static `fromPattern` method.

File: pomo.ts [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    const periods = parsePattern(options.pattern, options.scale ?? 1);
+    const cycle = new Cycle(periods);
+    return new Pomo(cycle, options.dayLength, options.ref);
+  }
+}
+
+export class PomoStamp {
```
Comment (EthanThatOneKid) [inline pomo.ts:63]: Provide brief JSDoc comment for the exported `PomoStamp` class.

File: .github/workflows/publish.yaml [hunk 8]
```diff
@@ -0,0 +1,40 @@
+name: publish
+
+on:
+  push:
+    branches:
+      - main
```
Comment (EthanThatOneKid) [inline .github/workflows/publish.yaml]: Use square brackets for list.
```suggestion
    branches: [main]
```

File: tasks/build_npm.ts [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  testPattern: "**/*_test.ts",
+  outDir: "./npm",
+  // see JS docs for overview and more options
+  shims: {
+    deno: true,
+  },
+  package: {
+    // package.json properties
+    name: "pomo",
+    version: VERSION,
```
Comment (EthanThatOneKid) [inline tasks/build_npm.ts]: Get the version string from the first argument.
```suggestion
    version: Deno.args[0],
```

File: tasks/build_npm.ts [hunk 10]
```diff
@@ -0,0 +1,33 @@
+import { build, emptyDir } from "https://deno.land/x/dnt@0.33.0/mod.ts";
```
Comment (EthanThatOneKid) [inline tasks/build_npm.ts]: Leave a note for the command intended to run this file.

File: tasks/build_npm.ts (hunk 9 above)
Comment (EthanThatOneKid) [reply tasks/build_npm.ts]: Also, be sure to edit `.github/workflows/publish.yaml` and pass the `$GITHUB_SHA` as the version string.

File: cycle.ts (hunk 1 above)
Comment (EthanThatOneKid) [reply cycle.ts:4]: Resolved by <https://github.com/EthanThatOneKid/pomo/pull/1/commits/cd57891ba330d047a91a76986a7c5c98c90f48ee>.

File: pomo.ts (hunk 3 above)
Comment (EthanThatOneKid) [reply pomo.ts:8]: Resolved by <https://github.com/EthanThatOneKid/pomo/pull/1/commits/5dd36d850714b6045bca3c5055b405409f5549b6>.

File: pomo.ts (hunk 4 above)
Comment (EthanThatOneKid) [reply pomo.ts]: Also renamed `infinite` to `eternal`. Resolved by <https://github.com/EthanThatOneKid/pomo/pull/1/commits/5dd36d850714b6045bca3c5055b405409f5549b6>.

File: pomo.ts (hunk 5 above)
Comment (EthanThatOneKid) [reply pomo.ts:30]: Resolved by <https://github.com/EthanThatOneKid/pomo/commit/5dd36d850714b6045bca3c5055b405409f5549b6>.

File: pomo.ts (hunk 6 above)
Comment (EthanThatOneKid) [reply pomo.ts:53]: Resolved in <https://github.com/EthanThatOneKid/pomo/commit/5dd36d850714b6045bca3c5055b405409f5549b6>.

<https://github.com/EthanThatOneKid/pomo/blob/5dd36d850714b6045bca3c5055b405409f5549b6/pomo.ts#L39-L50>

File: README.md (hunk 2 above)
Comment (EthanThatOneKid) [reply README.md]: Resolved by <https://github.com/EthanThatOneKid/pomo/pull/1/commits/c345788fd05726b461e690d6974affc486e61bf7>.

File: pomo.ts [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+    this.previousIndex = pomo.cycle.next(this.index, -1);
+  }
+
+  /** Whether or not the period is a work period. */
+  public get work(): boolean {
+    return this.index % 2 === 0;
+  }
+
+  /** The number remaining until the next period. */
+  public get until(): number {
```
Comment (EthanThatOneKid) [inline pomo.ts]: Nit: Consider renaming `until` to `timeout`.

File: tasks/build_npm.ts (hunk 9 above)
Comment (EthanThatOneKid) [reply tasks/build_npm.ts]: Resolved by <https://github.com/EthanThatOneKid/pomo/pull/1/commits/e263226f5abbd1ce5d73c83cbceaff05a7b6886b>.

File: pomo.ts (hunk 7 above)
Comment (EthanThatOneKid) [reply pomo.ts:63]: Resolved by <https://github.com/EthanThatOneKid/pomo/pull/1/commits/78b6a6a186b43d5215020e39418aa12c22aba4e0>.

File: tasks/build_npm.ts (hunk 10 above)
Comment (EthanThatOneKid) [reply tasks/build_npm.ts]: Resolved by <https://github.com/EthanThatOneKid/pomo/pull/1/commits/e263226f5abbd1ce5d73c83cbceaff05a7b6886b>.

File: pomo.ts (hunk 11 above)
Comment (EthanThatOneKid) [reply pomo.ts]: Resolved by <https://github.com/EthanThatOneKid/pomo/pull/1/commits/53bd79b1eb5fa07ef12742ee39cb5450b57f6859>.

**Review by boushrabettir — APPROVED** [review]
Good job!

**Pull request merged**